torch/csrc/jit/type.h

```diff
@@ -619,10 +647,20 @@ inline TypePtr CompleteTensorType::fromNumberType(TypePtr typ) {
     return CompleteTensorType::create(at::kLong, -1, {});
   } else if (typ->isSubtypeOf(FloatType::get())) {
     return CompleteTensorType::create(at::kFloat, -1, {});
+  } else if (typ->isSubtypeOf(BoolType::get())) {
+    return CompleteTensorType::create(at::kLong, -1, {});
   }
   AT_ERROR("unknown number type", typ->str());
 }
 
+inline TypePtr CompleteTensorType::fromBoolType(TypePtr type) {
```

Comment:
Why does this take `type` just to assert that it a `BoolType()`? Can't it just have 0 arguments?

Reply:
If nothing checks then it could end up with a failure in one of the `IValue::to...` methods at runtime versus failing here with some more helpful info

Reply:
That check should be happening in the caller of this function (where it is possible for it to be the wrong type). Otherwise future users are going to write `fromBoolType(BoolType::get())` which is weird.